libs/blocks/marketo-auto-config/marketo-auto-config.js

```diff
@@ -0,0 +1,226 @@
+    testInputs.forEach((input) => {
+      if (!input.value) {
+        inputValuesFilled = false;
+        if (requiredPanelExpandButton) {
+          requiredPanelExpandButton.click();
+        }
+        input.focus();
+      }
+    });
+    return inputValuesFilled;
+  };
+
+  const getUrl = () => {
+    const url = window.location.href.split('#')[0];
+    return `${url}#${utf8ToB64(JSON.stringify(stateReform(state)))}`;
+  };
+
+  const copyConfig = () => {
+    setConfigUrl(getUrl());
+    if (!navigator?.clipboard) {
+      setStatus(setIsError);
+      setShowConfigUrl(false);
+      return;
+    }
+    if (!configFormValidation()) {
+      setErrorMessage('Required fields must be filled');
+      setStatus(setIsError);
+      setShowConfigUrl(false);
+      return;
+    }
+
+    const linkTitle = document.getElementById('title').value;
+    const formId = document.getElementById('form_id').value;
+    const link = document.createElement('a');
+    link.href = getUrl();
+    link.textContent = `Marketo Form - ${linkTitle || formId}`;
+
+    const blob = new Blob([link.outerHTML], { type: 'text/html' });
+    // eslint-disable-next-line no-undef
+    const data = [new ClipboardItem({ [blob.type]: blob })];
+    navigator.clipboard.write(data)
+      .then(() => {
+        setStatus(setIsSuccess);
+        setErrorMessage('Failed to copy.');
+        setShowConfigUrl(true);
+      }, () => {
+        setStatus(setIsError);
+        setShowConfigUrl(false);
+      });
+  };
+
+  return html`
+  <textarea class=${`config-url ${showConfigUrl ? '' : 'hide'}`}>${configUrl}</textarea>
+  <button
+    class="copy-config"
+    onClick=${copyConfig}>Copy</button>
+  <div class="copy-message ${isError === true ? 'is-error' : ''} ${isSuccess === true ? 'is-success' : ''}">
+    <div class="success-message">Copied to clipboard!</div>
+    <div class="error-message">${errorMessage}</div>
+  </div>`;
+};
+
+const Input = ({ label, type = 'text', prop, placeholder }) => {
+  const context = useContext(ConfiguratorContext);
+  const onInputChange = (e) => {
+    context.dispatch({
+      type: 'INPUT_CHANGE',
+      prop,
+      value: type === 'checkbox' ? e.target.checked : e.target.value,
+    });
+  };
+
+  const value = { [type === 'checkbox' ? 'checked' : 'value']: (context.state[prop]) };
+  return html` <div class="field">
+      <label for=${prop}>${label}</label>
+      <input type=${type} id=${prop} name=${prop} ...${value} placeholder=${placeholder} onChange=${onInputChange} />
+    </div>`;
+};
+
+const FieldsPanel = () => html`
+  <${Input} label="Title (Optional)" prop="title" />
+  <${Input} label="Description (Optional)" prop="description" />
+  <${Input} label="Error Message (Optional)" prop="error_message" />
+  <${Input} label="Form ID*" prop="form_id" />
+  <${Input} label="Base URL*" prop="base_url" />
+  <${Input} label="Munchkin ID*" prop="munchkin_id" />
+  <${Input} label="Destination URL*" prop="destination_url" />
```

Comment:
These should all be optional

Reply:
I have default value for them, but I don't want users leave it as blank and grabbing data from backend. I think it is more clear giving them some highlights since they are core values to populate the form.